### ISSUE TITLE
realpath: /bitnami/apache/conf: No such file or directory

### ISSUE THREAD

**Issue by n-Hariri** [bitnami/bitnami-docker-wordpress#330]
**Description**

since the last update for `bitnami/wordpress `from debian 5.8.1 to  debian 5.8.2 ,containers fail to start with the error 
`realpath: /bitnami/apache/conf: No such file or directory`

I suspect that the path is missing /opt at the start because there is no apache folder under /bitnami anyway .

![image](https://user-images.githubusercontent.com/94356398/141943446-dbe1fd8a-fe9a-48f2-9c65-3527c53b3b2c.png)

It looks like there are some recent changes to `libfs.sh` and as I can see here 

https://github.com/bitnami/bitnami-docker-wordpress/commit/a959b009e34c86b4c570320e6b6bb3d111421d94#diff-60e73a5b6272846df361063f16054140244d41323cd80630b06c1dbaac49b609

I've tried to investigate but I'm not a batch expert and I don't know if those changes could have impacted the way the path is constructed.

We haven't changed any options in the docker-compose file so I guess it has to do with new image .

Thanks for investigating

**Comment by juan131** [comment]
Hi @n-Hariri 

Could you please the exact steps to follow to reproduce the issue? Thanks in advance

**Comment by n-Hariri** [comment]
Hi juan131,
We are running a stack using the following docker-compose.yml file 

```
version: '2'
services:
  mariadb_test:
    restart: unless-stopped
    image: docker.io/bitnami/mariadb:10.5
    user: xxxx
    volumes:
      - 'mariadb_data:/bitnami/mariadb'
    environment:
      - MARIADB_ROOT_PASSWORD=xxxxx
      - MARIADB_SKIP_TEST_DB=yes
      - MARIADB_USER=xxx
      - MARIADB_PASSWORD=xxxxx
      - MARIADB_DATABASE=xxx
    networks:
      - proxy
  web_test:
    image: docker.io/bitnami/wordpress:5
    user: xxxx
    volumes:
      - 'wordpress_data:/bitnami/wordpress'
    depends_on:
      - mariadb_test
    environment:
      # Database connection configuration
      - MARIADB_HOST=xxx
      - MARIADB_PORT_NUMBER=3306
      - WORDPRESS_DATABASE_USER=xxx
      - WORDPRESS_DATABASE_PASSWORD=xxxxxxx
      - WORDPRESS_DATABASE_NAME=xxxx
      - HTTP_PROXY=http://proxyserver:port
      - HTTPS_PROXY=http://proxyserver:port
      - NO_PROXY=localhost,127.0.0.1
    networks:
      - proxy
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=proxy"
      - "traefik.http.routers.web_test.entrypoints=websecure"
      - "traefik.http.routers.web_test.rule=Host(`test.domain`)"
      - "traefik.http.routers.web_test.tls=true"
volumes:
  mariadb_data:
  wordpress_data:
networks:
  # enable connection with Traefik
  proxy:
    external: true
```

**Comment by n-Hariri** [comment]
To make things clearer will provide you with 2 docker-compose files the only difference between both of them is the used bitnami wordpress image .

- the first one use image(works as expected )

>  bitnami/wordpress:5.8.1-debian-10-r58
	docker-compose.yml 

```
version: '2'
services:
  mariadb_test:
    restart: unless-stopped
    image: docker.io/bitnami/mariadb:10.5
    user: 'xxxx'
    volumes:
      - 'mariadb_data:/bitnami/mariadb'
    environment:
      - MARIADB_ROOT_PASSWORD=Rootpassword
      - MARIADB_SKIP_TEST_DB=yes
      - MARIADB_USER=test
      - MARIADB_PASSWORD=db_password
      - MARIADB_DATABASE=test
    networks:
      - proxy
  web_test:
    image: docker.io/bitnami/wordpress:5.8.1-debian-10-r58
    user: 'xxxxx'
    volumes:
      - 'wordpress_data:/bitnami/wordpress'
    depends_on:
      - mariadb_test
    environment:
      # Database connection configuration
      - MARIADB_HOST=mariadb_test
      - MARIADB_PORT_NUMBER=3306
      - WORDPRESS_DATABASE_USER=test
      - WORDPRESS_DATABASE_PASSWORD=db_password
      - WORDPRESS_DATABASE_NAME=test
      #- WORDPRESS_SKIP_INSTALL=yes
      - WORDPRESS_ENABLE_HTACCESS_PERSISTENCE=yes
      - WORDPRESS_HTACCESS_OVERRIDE_NONE=yes
      - HTTP_PROXY=http://proxy.domain:port
      - HTTPS_PROXY=http://proxy.domain:port
      - NO_PROXY=localhost,127.0.0.1
    networks:
      - proxy
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=proxy"
      - "traefik.http.routers.web_test.entrypoints=websecure"
      - "traefik.http.routers.web_test.rule=Host(`test.domain`)"
      - "traefik.http.routers.web_test.tls=true"
volumes:
  mariadb_data:
  wordpress_data:
networks:
  # enable connection with Traefik
  proxy:
    external: true
```
the stack run as expected with the following logs 
```

wordpress 14:36:59.79 ,
wordpress 14:36:59.79 Welcome to the Bitnami wordpress container,
wordpress 14:36:59.79 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-wordpress,
wordpress 14:36:59.79 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-wordpress/issues,
wordpress 14:36:59.79 ,
wordpress 14:36:59.80 INFO  ==> ** Starting WordPress setup **,
realpath: /bitnami/apache/conf: No such file or directory,
wordpress 14:36:59.85 INFO  ==> Configuring PHP options,
wordpress 14:36:59.87 INFO  ==> Validating settings in MYSQL_CLIENT_* env vars,
wordpress 14:37:01.22 INFO  ==> Ensuring WordPress directories exist,
wordpress 14:37:01.22 INFO  ==> Trying to connect to the database server,
wordpress 14:37:06.25 INFO  ==> Configuring WordPress with settings provided via environment variables,
wordpress 14:37:08.64 INFO  ==> Installing WordPress,
wordpress 14:37:23.31 INFO  ==> Persisting WordPress installation,
wordpress 14:37:27.62 INFO  ==> ** WordPress setup finished! **,
wordpress 14:37:27.63 INFO  ==> ** Starting Apache **,
[Tue Nov 16 14:37:28.053093 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name,
[Tue Nov 16 14:37:28.053764 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name,
[Tue Nov 16 14:37:28.086383 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name,
[Tue Nov 16 14:37:28.086925 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name,
[Tue Nov 16 14:37:28.105545 2021] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.51 (Unix) OpenSSL/1.1.1d PHP/7.4.25 configured -- resuming normal operations,
[Tue Nov 16 14:37:28.105593 2021] [core:notice] [pid 1] AH00094: Command line: '/opt/bitnami/apache/bin/httpd -f /opt/bitnami/apache/conf/httpd.conf -D FOREGROUND',
```

- the second use the image

>  bitnami/wordpress:5

> docker-compose.yml

```
version: '2'
services:
  mariadb_test:
    restart: unless-stopped
    image: docker.io/bitnami/mariadb:10.5
    user: 'xxxx'
    volumes:
      - 'mariadb_data:/bitnami/mariadb'
    environment:
      - MARIADB_ROOT_PASSWORD=Rootpassword
      - MARIADB_SKIP_TEST_DB=yes
      - MARIADB_USER=test
      - MARIADB_PASSWORD=db_password
      - MARIADB_DATABASE=test
    networks:
      - proxy
  web_test:
    image: docker.io/bitnami/wordpress:5
    user: 'xxxxx'
    volumes:
      - 'wordpress_data:/bitnami/wordpress'
    depends_on:
      - mariadb_test
    environment:
      # Database connection configuration
      - MARIADB_HOST=mariadb_test
      - MARIADB_PORT_NUMBER=3306
      - WORDPRESS_DATABASE_USER=test
      - WORDPRESS_DATABASE_PASSWORD=db_password
      - WORDPRESS_DATABASE_NAME=test
      #- WORDPRESS_SKIP_INSTALL=yes
      - WORDPRESS_ENABLE_HTACCESS_PERSISTENCE=yes
      - WORDPRESS_HTACCESS_OVERRIDE_NONE=yes
      - HTTP_PROXY=http://proxy.domain:port
      - HTTPS_PROXY=http://proxy.domain:port
      - NO_PROXY=localhost,127.0.0.1
    networks:
      - proxy
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=proxy"
      - "traefik.http.routers.web_test.entrypoints=websecure"
      - "traefik.http.routers.web_test.rule=Host(`test.domain`)"
      - "traefik.http.routers.web_test.tls=true"
volumes:
  mariadb_data:
  wordpress_data:
networks:
  # enable connection with Traefik
  proxy:
    external: true
```
the wordpress container fail to start with following logs
 
```
wordpress 15:22:52.96 ,
wordpress 15:22:52.96 Welcome to the Bitnami wordpress container,
wordpress 15:22:52.96 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-wordpress,
wordpress 15:22:52.96 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-wordpress/issues,
wordpress 15:22:52.97 ,
wordpress 15:22:52.97 INFO  ==> ** Starting WordPress setup **,
realpath: /bitnami/apache/conf: No such file or directory,
wordpress 15:22:53.01 INFO  ==> Configuring PHP options,
wordpress 15:22:53.03 INFO  ==> Validating settings in MYSQL_CLIENT_* env vars,
cp: cannot stat '/opt/bitnami/apache/conf/vhosts/htaccess/wordpress-htaccess.conf': No such file or directory,

```
In addition we tried image tag 5.8.2 and worked as expected although 5 and 5.8.2 both have the same digest as shown below .

As a result I think the real path error is not the reason why our containers are broken It's something different. 

Appreciate your help .
Thanks

**Comment by juan131** [comment]
Hi @n-Hariri 

Are you able to reproduce the issue when there's no persisted data in the volume you're mounting in the WP container? or does it only happen when restarting the container with previous data?

**Comment by n-Hariri** [comment]
Hi Juan131,

After investigating more We found out what is the problem 
when using image tag 5 it was using the existing image which is pulled a week ago when our containers broke and every time we ran docker-compose for image tag 5 it wasn't pulling the latest which explain why we had different behavior between 5 and 5.8.2 .
 so now the containers run on 5.8.2 but we still don't know what happened a week ago when they all broke .

I guess a fix has been implemented in the next release ?

to answer your question using the old tag 5 image the issue was happening in both cases with persisted data in the volumes and without .

Thank you for your help .

**Comment by juan131** [comment]
Thanks for the context @n-Hariri 

It's weird that a previous version broke the persistence setup since we do have tests in our internal pipeline that ensure the scenario below is working as expected:

- We launch a WordPress container with persistence.
- After that, we perform some basic "write actions" (e.g. creating a new post)
- Next, we restart the container, and test everything is working and the data was persisted.
- Finally, we recreate the container, and test again that everything is working and the data is there.

No image should have been released if the scenario described above fails.

In any case, I'm very glad you were able to solve the issue using the latest image. I proceed to close the issue as "solved" but do not hesitate to reopen it if you require further assistance.
